### PR TITLE
Task 9

### PR DESCRIPTION
**Pull request by yrakova**
Description

Write unit tests for your application (consider using Jest, @testing-library/react or Enzyme, react-test-renderer, React-test-utils, etc.). 

Subtasks: 
- [x] 1. Cover 1 simple presentational component with snapshot tests; 
- [x] 2. Cover 1 reducer and all its actions with unit-tests; 
- [x] 3. Measure coverage level with coverage report; 
- [x] 4. Cover “Add movie” modal dialog components with unit-tests, mock all external dependencies using Jest mocks. 
 
Evaluation criteria* 
 
- [x] 2 - Subtask 1 is implemented; 
- [x] 3 - Subtasks 2,3 are implemented; 
- [x] 4 - Subtask 4 “Add movie” modal dialog and all its components coverage > 70%; 
- [ ] 5 - Global coverage > 90%; add unit tests for hooks. 
 
Each mark includes previous mark criteria. 

### [Task-9-Live-Preview](https://607fbd9a0abd9300078dfd81--yrakova-spa-movies.netlify.app/)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,5 +75,16 @@
       "npm run eslint-check",
       "git add"
     ]
+  },
+  "jest": {
```
Comment (mmarkelov) [inline package.json]: Not an error, but it's better to put jest configuration in separate file

File: src/__tests__/AddMovieModal.test.jsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    store = mockStore(stateInitial);
+  });
+  test('AddMovie is shown', () => {
+    const { queryByText, queryByRole } = render(
+      <AddMovieModal show={true} onAction={mockAction} />,
+      {
+        wrapper: ({ children }) => (
+          <Provider store={store}>{children}</Provider>
+        ),
+      }
```
Comment (mmarkelov) [inline src/__tests__/AddMovieModal.test.jsx]: I think that you can move it to separate function to reuse it. Like this:
```js
const getWrapper = () => ({
         wrapper: ({ children }) => (
          <Provider store={store}>{children}</Provider>
        ), 
})
```

File: src/__tests__/AddMovieModal.test.jsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    );
+    const addMovieModal = queryByText('Add Movie');
+    expect(addMovieModal).toBeNull();
+  });
+
+  test('Edit Movie Modal is hidden', () => {
+    const mockMovie = mockMovies[0];
```
Comment (mmarkelov) [inline src/__tests__/AddMovieModal.test.jsx]:
```js
const [mockMovie] = mockMovies;
```

File: src/__tests__/AddMovieModal.test.jsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      expect(screen.getByLabelText(/Title/i)).toHaveValue(mockMovie.title);
+      expect(screen.getByLabelText(/Poster URL/i)).toHaveValue(mockMovie.poster_path);
+      expect(screen.getByLabelText(/Runtime/i)).toHaveValue(mockMovie.runtime);
+      expect(screen.getByLabelText(/Overview/i)).toHaveValue(mockMovie.overview);
+      expect(screen.getByLabelText(/Release Date/i)).toHaveValue(mockMovie.release_date);      
+      expect(screen.queryByTestId(/genres/i).children[1]).toHaveTextContent(mockMovie.genres.join(''));      
+    });
+
+    test('SAVE success', async () => {
+      const newTitle = mockMovie.title + ' 123';
```
Comment (mmarkelov) [inline src/__tests__/AddMovieModal.test.jsx]:
```js
const newTitle = `${mockMovie.title} 123`;
```

File: package.json (hunk 1 above)
Comment (yrakova) [reply package.json]: Ok, thanks

File: src/__tests__/AddMovieModal.test.jsx (hunk 2 above)
Comment (yrakova) [reply src/__tests__/AddMovieModal.test.jsx]: Ok, thanks

File: src/__tests__/AddMovieModal.test.jsx (hunk 4 above)
Comment (yrakova) [reply src/__tests__/AddMovieModal.test.jsx]: Ok, thanks

File: src/__tests__/AddMovieModal.test.jsx (hunk 3 above)
Comment (yrakova) [reply src/__tests__/AddMovieModal.test.jsx]: Sure, thanks!

**Pull request merged**